src/java.base/share/classes/jdk/internal/constant/MethodTypeDescImpl.java

```diff
@@ -105,18 +105,46 @@ public static MethodTypeDescImpl ofValidated(ClassDesc returnType, ClassDesc...
      * @jvms 4.3.3 Method Descriptors
      */
     public static MethodTypeDescImpl ofDescriptor(String descriptor) {
-        // Implicit null-check of descriptor
-        List<ClassDesc> ptypes = ConstantUtils.parseMethodDescriptor(descriptor);
```

Comment:
This might have been the only use of parseMethodDescriptor - so you can probably remove that method, making this patch a net clean-up in terms of lines of code.